frontend/src/components/Collections/components/Grid/components/Row/DatasetRow/components/UploadStatus/index.tsx

```diff
@@ -1,37 +1,65 @@
-import { Intent, Spinner } from "@blueprintjs/core";
+import { Button, Intent } from "@blueprintjs/core";
 import React, { FC } from "react";
-import { CancelButton, DatasetStatusTag, StatusContainer } from "./style";
+import DeleteDataset from "../DeleteDataset";
+import { DatasetStatusTag, StatusContainer, StyledSpinner } from "./style";
 
 interface Props {
+  isConverting: boolean;
   isValidating: boolean;
   progress: number;
-  cancelUpload: () => void;
+  datasetId: string;
+  collectionId: string;
 }
 
-const UploadStatus: FC<Props> = ({ isValidating, progress, cancelUpload }) => {
-  const cancel = <CancelButton onClick={cancelUpload}>Cancel</CancelButton>;
+const UploadStatus: FC<Props> = ({
+  isConverting,
+  isValidating,
+  progress,
+  datasetId,
+  collectionId,
+}) => {
+  let content: { progress?: number; text: string } = {
+    progress,
+    text: `Uploading (${Math.round(progress * 100)}%)`,
+  };
+
+  if (isConverting) {
+    content = {
+      text: "Processing...",
+    };
+  }
 
   if (isValidating) {
-    return (
```

Comment:
Keeping the code DRY by removing the duplicate code and consolidating the changing content in `content` object